python/cuml/solvers/sgd.pyx

```diff
@@ -443,7 +453,9 @@ class SGD(Base):
 
         del(X_m)
 
-        return preds
+        output_type = self._get_output_type(self.output_type)
```

Comment:
```suggestion
        output_type = self._get_output_type(X)
```

To mirror things correctly, the parameter here should be `X` as opposed to `self.output_type`